src/content/reference/react/useInsertionEffect.md

```diff
@@ -22,81 +22,82 @@ useInsertionEffect(setup, dependencies?)
 
-#### Caveats {/*caveats*/}
+#### Avisos {/*caveats*/}
+
+* Effects (efeitos) só são executados pelo cliente. Eles não são executados durante a renderização do servidor.
+* Não é possível atualizar o estado de dentro do `useInsertionEffect`.
+* No momento em que `useInsertionEffect` é executado, as refs ainda não foram anexadas.
+* O `useInsertionEffect` pode ser utilizado antes ou depois do DOM ter sido atualizado. Você não deve confiar que o DOM seja atualizado em um determinado momento.
+* Ao contrário de outros tipos de Effects, que disparam a limpeza para cada Effect e depois a configuração para cada Effect, `useInsertionEffect` irá disparar tanto a limpeza quanto a configuração de um componente de cada vez. Isso resulta em uma “intercalação” das funções de limpeza e configuração.
 
-* Effects only run on the client. They don't run during server rendering.
-* You can't update state from inside `useInsertionEffect`.
-* By the time `useInsertionEffect` runs, refs are not attached yet.
-* `useInsertionEffect` may run either before or after the DOM has been updated. You shouldn't rely on the DOM being updated at any particular time.
-* Unlike other types of Effects, which fire cleanup for every Effect and then setup for every Effect, `useInsertionEffect` will fire both cleanup and setup one component at a time. This results in an "interleaving" of the cleanup and setup functions.
 ---
 
-## Usage {/*usage*/}
+## Utilização {/*usage*/}
 
-### Injecting dynamic styles from CSS-in-JS libraries {/*injecting-dynamic-styles-from-css-in-js-libraries*/}
+### Injetar os estilos dinâmicos em bibliotecas CSS-em-JS {/*injecting-dynamic-styles-from-css-em-js-libraries*/}
 
-Traditionally, you would style React components using plain CSS.
+Normalmente, os componentes React são estilizados com CSS simples.
 
 ```js
-// In your JS file:
+// Em seu arquivo JS:
 <button className="success" />
 
-// In your CSS file:
+// Em seu arquivo CSS:
 .success { color: green; }
 ```
 
-Some teams prefer to author styles directly in JavaScript code instead of writing CSS files. This usually requires using a CSS-in-JS library or a tool. There are three common approaches to CSS-in-JS:
+Algumas equipes preferem criar estilos diretamente no código JavaScript em vez de escrever arquivos CSS. Isso geralmente requer o uso de uma biblioteca CSS-em-JS ou de alguma ferramenta. Existem três abordagens comuns para CSS-em-JS:
 
-1. Static extraction to CSS files with a compiler
-2. Inline styles, e.g. `<div style={{ opacity: 1 }}>`
-3. Runtime injection of `<style>` tags
+1. A extração estática para os arquivos CSS por um compilador
+2. Estilos em linha, por exemplo, `<div style={{ opacity: 1 }}>`
+3. Injeção em tempo de execução de tags `<style>`
 
-If you use CSS-in-JS, we recommend a combination of the first two approaches (CSS files for static styles, inline styles for dynamic styles). **We don't recommend runtime `<style>` tag injection for two reasons:**
```

Comment:
Original: we recommend a combination of the first two approaches
 
Seria interessante: Nós recomendamos a combinação das duas primeiras abordagens.

Como você fez no segundo paragrafo.

Reply:
A frase "É recomendável uma combinação das duas primeiras abordagens" é mais fluida em português. Ela mantém a mesma ideia da frase original em inglês de forma mais natural e direta.